radix-engine/src/engine/call_frame.rs

```diff
@@ -1577,15 +1596,29 @@ where
                         .cloned()
                         .ok_or(RuntimeError::ValueNotFound(ValueId::Vault(*vault_id)))?
                 };
+                if is_pay_fee && !matches!(cur_location, REValuePointer::Track { .. }) {
```

Comment:
Could just have the full `is_pay_fee` check here.

Reply:
Commented in another thread.